core/arch/asm_defines.asm

```diff
@@ -910,27 +910,71 @@ ASSUME fs:_DATA @N@\
         mov      ARG1, p1   @N@\
         blx      callee
 #elif defined(RISCV64)
+/* For RISC-V, there is no a instruction can operate on both immediates
+ * and registers. Here is a macro that judges whether its argument is
+ * a register or not.
+ */
+/* TODO i#3544: Temporary solution */
```

Comment:
Is this temporary?  Do we have a better solution in mind?  If we don't have a better one I would think this is our long-term solution and we'd just remove this comment.